cmd/file.d.go

```diff
@@ -85,23 +87,35 @@ func start() {
 	fileD.Start()
 }
 
-func listenSignals() {
+func listenSignals(ctx context.Context) {
 	signalChan := make(chan os.Signal)
 	signal.Notify(signalChan, syscall.SIGHUP, syscall.SIGTERM)
 
 	for {
+		ctx, cancel := context.WithCancel(ctx)
 		s := <-signalChan
 
 		switch s {
 		case syscall.SIGHUP:
 			logger.Infof("SIGHUP received")
-			fileD.Stop()
+			fileD.Stop(ctx)
+			go func() {
+				time.Sleep(200 * time.Millisecond)
+				cancel()
+			}()
+			time.Sleep(200 * time.Millisecond)
+
```

Comment:
what happens if we sent cancel(), stop process fails, and then we say start? could there be problems?

Reply:
thank for notion, a fix this.